### ISSUE TITLE
Fork indicator that does not take up space but expands 

### ISSUE THREAD

**Issue by aerugo** [Edgeryders-Participio/babel-reader#9]
We need a fork indicator that does not take up space between the paragraphs. The reason for this is that the reader loads all threads in the background and then starts looking through incoming links for forks. When a fork is found, a fork indicator is added to the right place. If the fork indicator takes up more space than the regular space between paragraphs, the reader can start pushing your view around when you are reading as the text is shifted to make room for the fork indicators.

At the same time, we don't want the whitespace between paragraphs to be larger for between posts than it is within them as it disrupts the smooth reading experience.

One possible solution would be to have a fork indicator that is very small and half-faded so that it does not take up too much mind space. See image below:

![Screenshot 2020-05-27 at 15 35 31](https://user-images.githubusercontent.com/7785081/83028583-ebe95300-a031-11ea-9c24-37a444c15eed.jpg)

This would then animate to expand into the selector.

@skoteskote what is your take on this?

**Comment by skoteskote** [comment]
>  If the fork indicator takes up more space than the regular space between paragraphs, the reader can start pushing your view around when you are reading as the text is shifted to make room for the fork indicators.

Wouldn't it only push the view down below the last post before the fork, which wouldn't affect the reader? 
I have nothing against your solution, I think it looks good, and if this is the easiest let's go with it. In an ideal situation I'd make the fork space bigger than regular paragraph space, to highlight that the fork is something extra. But if that is difficult then let's go with your suggestion.

**Comment by aerugo** [comment]
> Wouldn't it only push the view down below the last post before the fork, which wouldn't affect the reader?

No. What can happen is this:

1) You load a fork directly, this throws you right into the middle of the long thread with threads before it. As the reader starts validating incoming links to look for forks, it may very well find more forks from posts above where you currently are.

2) You load a topic that is not itself a fork but immediately start scrolling and read a part further down, which is then pushed down in front of you as forks are loaded above.

Granted, case 2 will be quite rare, but case 1 will be quite common.

**Comment by skoteskote** [comment]
Ah I see, go with the small fork symbol solution then!

**Comment by aerugo** [comment]
@jims, what do you think about that solution?

**Comment by jims** [comment]
Yeah as long as it fits between paragraph margins its fine. We would also make the actual selection be an overlay then I suppose. On a side note, I'm not sure how much this will be an issue in practice actually, It might be that it is a very rare case that this actually happens.
